argo_jupyter_scheduler/executor.py

```diff
@@ -573,3 +663,149 @@ def create_job_record(
+
+        old_output_path = gen_default_output_path(input_path)
+        old_html_path = gen_default_html_path(input_path)
+
+        new_output_path = gen_output_path(input_path, start_time)
+        new_html_path = gen_html_path(input_path, start_time)
+
+        # Renames files
+        os.rename(old_output_path, new_output_path)
+        os.rename(old_html_path, new_html_path)
+
+        logger.info("Successfully renamed files")
+
+    except Exception as e:
+        msg = "Failed to rename files"
+        logger.info(msg)
+        raise Exception(msg) from e
+
+
+def get_slack_token_channel(parameters):
+    token = None
+    channel = None
+
+    if parameters is not None:
+        token = parameters.get("SLACK_TOKEN")
+        channel = parameters.get("SLACK_CHANNEL")
+
+    return token, channel
+
+
+@script()
+def send_to_slack(
+    db_url, job_definition_id, input_path, start_time, token, channel, log_path
+):
+    import json
+
+    import requests
+    from jupyter_scheduler.orm import Job, create_session
+
+    from argo_jupyter_scheduler.utils import (
+        add_file_logger,
+        gen_html_path,
+        gen_timestamp,
+        setup_logger,
+    )
+
+    try:
+        # Sets up logging
+        logger = setup_logger("send_to_slack")
+        add_file_logger(logger, log_path)
+
+        # Gets start_time if not provided to generate file path
+        if start_time is None:
+            db_session = create_session(db_url)
+            with db_session() as session:
+                q = (
+                    session.query(Job)
+                    .filter(Job.job_definition_id == job_definition_id)
+                    .order_by(Job.start_time.desc())
+                    .first()
+                )
+
+                start_time = q.start_time
+
+        start_time = gen_timestamp(start_time)
+
+        html_path = gen_html_path(input_path, start_time)
+
+        # Sends to Slack
+        url = "https://slack.com/api/files.upload"
+
+        files = {"file": (os.path.basename(html_path), open(html_path, "rb"))}
+
+        data = {
+            "initial_comment": "Attaching new file",
+            "channels": channel,
+        }
+
+        headers = {
+            "Authorization": f"Bearer {token}",
+        }
+
+        logger.info(f"Sending to Slack: file: {html_path}, channel: {channel}")
+        response = requests.post(
+            url, files=files, data=data, headers=headers, timeout=30
+        )
+        response.raise_for_status()
+
+        response = response.text
+        logger.info(f"Slack response: {response}")
+
+        if not json.loads(response).get("ok"):
+            msg = "Unexpected response when sending to Slack"
+            logger.info(msg)
+            raise Exception(msg)
+
+        logger.info("Successfully sent to Slack")
+
+    except Exception as e:
+        msg = "Failed to send to Slack"
+        logger.info(msg)
+        raise Exception(msg) from e
```

Comment:
This now uses the requests library. Added logging to stdout and a file. Wraps everything into try/except to give more context on where the error occurs. Similar code in the rename script.